Modules/mmapmodule.c

```diff
@@ -312,36 +385,67 @@ mmap_read_byte_method(mmap_object *self,
     }
 }
 
+PyObject *
+_read_mmap_mem(mmap_object *self, char *start, size_t num_bytes) {
+    if (num_bytes == 1) {
+        char dest;
+        if (safe_byte_copy(&dest, start) < 0) {
+            return NULL;
+        }
+        else {
+            PyBytesObject *op = CHARACTER(dest & 255);
```

Comment:
Is having this here really an improvement over `PyBytes_FromStringAndSize(&dest, 1)`? Performance wise, probably, but I don't like having performance/memory optimizations like this being replicated throughout the codebase.

Reply:
Since this optimization is present in `PyBytes_FromStringAndSize` https://github.com/python/cpython/blob/8227883d1f1bbb6560e5f175d7ee49f013c094bd/Objects/bytesobject.c#L120-L123 but is never used when passing NULL I added it here since you mentioned above that single byte performance is important in regard to memcpy.

Reply:
Agreed, I'm proposing passing `&dest` instead of `NULL` so we'll still get the optimisation but without duplicating the code.

Maintainability is also important, as well as performance. And the performance will always be better for callers who read more than one byte at a time, so unless we're talking a 50-100% difference on a microbenchmark, I think maintainability wins (the `NULL` behaviour includes dynamic memory allocation, so is _definitely_ going to be much slower than passing `&dest`).

Reply:
@zooba Please have a look at my latest commit if that's the way you meant to implement it.